gnovm/pkg/gnolang/op_assign.go

```diff
@@ -46,7 +46,7 @@ func (m *Machine) doOpAddAssign() {
 	// add rv to lv.
 	addAssign(m.Alloc, lv.TV, rv)
 	if lv.Base != nil {
-		m.Realm.DidUpdate(lv.Base.(Object), nil, nil)
+		m.Realm.DidUpdate(m.Store, lv.Base.(Object), nil, nil)
```

Comment:
I don't think it should be necessary to pass store in for inter-realm logic.
There's nothing in the spec that would necessitate loading anything from the store.
https://docs.google.com/document/d/1eCgGPCJ8fMhNc_vc_pbgCxP10X7jGKLC89nBUGatsD4/edit?tab=t.0

If something is in the store it is a RefValue, and from the RefValue you can tell what realm it's attached (stored) to. No need to load anything. 

Probably because it's trying to do things from the RHS, and not purely from the LHS, as described here: https://github.com/gnolang/gno/pull/2958/files#r1986207268

I'm making another PR to demonstrate a simpler impl.